src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/task/VisitTask.kt

```diff
@@ -0,0 +1,44 @@
+package uk.gov.justice.digital.hmpps.visitscheduler.task
+
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.visitscheduler.data.filter.VisitFilter
+import uk.gov.justice.digital.hmpps.visitscheduler.jpa.VisitStatus
+import uk.gov.justice.digital.hmpps.visitscheduler.service.VisitSchedulerService
+import java.time.LocalDateTime
+
+@Component
+class VisitTask(
+  private val visitSchedulerService: VisitSchedulerService,
+  @Value("\${task.expired-visit.enabled:false}") private val enabled: Boolean,
+  @Value("\${task.expired-visit.validity-minutes:20}") private val expiredPeriod: Long
+) {
+
+  @Scheduled(cron = "\${task.expired-visit.cron:0 0/15 * * * ?}")
```

Comment:
so some slots might have a live for up to 30 minutes? is that right?

Clean up at 12:00
Booking is made 12:00:01
Clean up at 12:15  (not out of date)
Clean up at 12:30 (visit expired)

Reply:
oh, it's overiden per env

Reply:
For now yes.  The actual schedule has not been defined and is configurable, It might be the case that we only want to clean up over night.  You are correct, to ensure expired reservations are cleaned out immediately or as close as possible you would need to trigger the event every minute - this may or may not have an impact on the DB performance depending on the number of available connections and number of records.   Once we have realistic metrics we can set the defaults to the appropriate levels which will not impact performance.